vscode/src/chat/agentic/DeepCody.ts

```diff
@@ -161,4 +257,86 @@ export class DeepCodyAgent extends CodyChatAgent {
             return []
         }
     }
+
+    protected async processStream(
+        requestID: string,
+        message: Message[],
+        signal?: AbortSignal,
+        model?: string
+    ): Promise<string> {
+        const stream = await this.chatClient.chat(
+            message,
+            { model, maxTokensToSample: 4000 },
+            new AbortController().signal,
+            requestID
+        )
+        const accumulated = new RawTextProcessor()
+        try {
+            for await (const msg of stream) {
+                if (signal?.aborted) break
+                if (msg.type === 'change') {
+                    const newText = msg.text.slice(accumulated.length)
+                    accumulated.append(newText)
+                    await this.multiplexer.publish(newText)
+                }
+
+                if (msg.type === 'complete' || msg.type === 'error') {
+                    if (msg.type === 'error') throw new Error('Error while streaming')
```

Comment:
This way, we probably lose valuable context about the error. Does the message have a field with the error message we can use?

```suggestion
                    if (msg.type === 'error') throw new Error('Error while streaming')
```

Reply:
update to use msg.error instead